src/texture.toit

```diff
@@ -554,54 +554,66 @@ abstract class TextTexture_ extends SizedTexture:
       string_ = new_string
       fix_bounding_box_
       return
-    left_doesnt_move  := alignment_ == TEXT_TEXTURE_ALIGN_LEFT
-    right_doesnt_move := alignment_ == TEXT_TEXTURE_ALIGN_RIGHT
+
+    get_bounding_boxes_ string_ new_string alignment_ font_: | changed_extent_old/TextExtent_ changed_extent_new/TextExtent_ |
+      invalidate_extent_ changed_extent_old
+      invalidate_extent_ changed_extent_new
+      string_ = new_string
+      fix_bounding_box_
+      return
+    string_ = new_string
+    update_
+
+  static get_bounding_boxes_ old/string new/string alignment/int font/Font [block]:
+    left_doesnt_move  := alignment == TEXT_TEXTURE_ALIGN_LEFT
+    right_doesnt_move := alignment == TEXT_TEXTURE_ALIGN_RIGHT
     // If the rendered length does not change then both ends don't move.
-    // for center alignment it gets a little harder to detect which part of the
-    // display to invalidate so we don't bother with this case for now.
-    if alignment_ != TEXT_TEXTURE_ALIGN_CENTER and (font_.pixel_width string_) == (font_.pixel_width new_string):
+    pixel_width_old := font.pixel_width old
+    if pixel_width_old == (font.pixel_width new):
       left_doesnt_move = true
       right_doesnt_move = true
-    l := min string_.size new_string.size
+    l := min old.size new.size
```

Comment:
Use a variable name longer than one char?
At the moment I'm still trying to figure out whether it stands for "left", or "length", or something else.
edit: I'm guessing it's "length".